src/seql/core.clj

```diff
@@ -258,12 +266,18 @@
    recompose a tree of entities as specified in fields.
 
    "
-  [fields records]
-  (letfn [(add-relation-fn [group]
+  [schema fields records]
+  (letfn [(rel-type-fn [schema rel-key]
+            (let [rel-namespace (-> rel-key namespace keyword)]
+              (get-in schema [rel-namespace :relations rel-key :type])))
```

Comment:
This could live inside `add-relation-fn`'s let, but I suppose it's harmless here as well. The shadowing of `schema` with itself is unnecessary though.

Reply:
Thanks  :) will move it to the let.